edit-feature-attachments/src/main/java/com/esri/arcgismaps/sample/editfeatureattachments/MainActivity.kt

```diff
@@ -0,0 +1,350 @@
+        lifecycleScope.launch {
+            mapView.onSingleTapConfirmed.collect { tapConfirmedEvent ->
+                // clear any previous selection
+                featureLayer.clearSelection()
+                // identify tapped feature
+                mapView.identifyLayer(
+                    layer = featureLayer,
+                    screenCoordinate = tapConfirmedEvent.screenCoordinate,
+                    tolerance = 5.0,
+                    returnPopupsOnly = false,
+                    maximumResults = 1
+                ).onSuccess { layerResult ->
+                    val resultGeoElements: List<GeoElement> = layerResult.geoElements
+                    // check if a feature was identified
+                    if (resultGeoElements.isNotEmpty() && resultGeoElements.first() is ArcGISFeature) {
+                        // retrieve and set the currently selected feature
+                        val selectedFeature = resultGeoElements.first() as ArcGISFeature
+                        // highlight the currently selected feature
+                        featureLayer.selectFeature(selectedFeature)
+
+                        // show the bottom sheet layout
+                        createBottomSheet(selectedFeature)
+
+                        // keep track of the selected feature
+                        selectedArcGISFeature = selectedFeature
+                    }
+                }.onFailure {
+                    showError("Failed to select feature: ${it.message}")
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates and displays a bottom sheet to display and modify
+     * the attachments of [selectedFeature]. Calls [AttachmentsBottomSheet] to
+     * inflate bottom sheet and listen for interactions.
+     */
+    private suspend fun createBottomSheet(selectedFeature: ArcGISFeature) {
+        // get the number of attachments
+        val attachmentList = selectedFeature.fetchAttachments().getOrElse {
+            return showError(it.message.toString())
+        }
+        // get the attribute "typdamage" of the selected feature
+        val damageTypeAttribute = selectedFeature.attributes["typdamage"].toString()
+
+        // creates a new BottomSheetDialog
+        bottomSheet = AttachmentsBottomSheet(
+            context = this@MainActivity,
+            bottomSheetBinding = attachmentsSheetBinding,
+            attachments = attachmentList,
+            damageType = damageTypeAttribute
+        )
+        // set the content view to the root of the binding layout
+        bottomSheet?.setContentView(attachmentsSheetBinding.root)
+        // display the bottom sheet view
+        bottomSheet?.show()
+    }
+
+    /**
+     * Retrieves the [attachment] data in the form of a byte array, converts it
+     * to a [BitmapDrawable], caches the bitmap as a png image, and open's the
+     * attachment image in the default image viewer.
+     */
+    fun fetchAttachmentAsync(attachment: Attachment) {
```

Comment:
I see the problem that the call site is actually the `init` of `AttachmentsBottomSheet`, which won't have access to a scope unless you pass it in. I'm not sure what the best option is in that case:

- pass a scope to the bottom sheet constructor
- make an initializing method on the bottom sheet class which is a suspend fun

In either case it does seem a bit like overkill for the purpose of this sample, so I leave it up to you whether this would just cloud the functionality

Reply:
I can launch a coroutine from the `init` of `AttatchmentsBottomSheet` as it extends the [BottomSheetDialog](https://developer.android.com/reference/com/google/android/material/bottomsheet/BottomSheetDialog). Removed the `async` term and updated the function to be a suspend fun.